src/snowflake/cli/plugins/nativeapp/commands.py

```diff
@@ -268,18 +270,24 @@ def app_teardown(
 def app_deploy(
     prune: Optional[bool] = typer.Option(
         default=None,
-        help=f"""Whether to delete specified files from the stage if they don't exist locally. If set, the command deletes files that exist in the stage, but not in the local filesystem.""",
+        help=f"""Whether to delete specified files from the stage if they don't exist locally. If set, the command deletes files that exist in the stage, but not in the local filesystem. This option cannot be used when paths are specified.""",
     ),
     recursive: Optional[bool] = typer.Option(
         None,
         "--recursive/--no-recursive",
         "-r",
         help=f"""Whether to traverse and deploy files from subdirectories. If set, the command deploys all files and subdirectories; otherwise, only files in the current directory are deployed.""",
     ),
-    files: Optional[List[Path]] = typer.Argument(
+    paths: Optional[List[Path]] = typer.Argument(
         default=None,
         show_default=False,
-        help=f"""Paths, relative to the the project root, of files you want to upload to a stage. The paths must match one of the artifacts src pattern entries in snowflake.yml. If unspecified, the command syncs all local changes to the stage.""",
+        help=dedent(
+            f"""
+            Paths, relative to the the project root, of files or directories you want to upload to a stage. If a file is
+            specified, it must match one of the artifacts src pattern entries in snowflake.yml. If a directory is
+            specified, it will be searched for subfolders or files to deploy based on artifacts src pattern entries. If
+            unspecified, the command syncs all local changes to the stage."""
+        ).strip(),
```

Comment:
iirc, the reason why this only says file or directory, and not a glob pattern, is because a user's shell should automatically expand that and pass that in to this arg as a list. Is that correct? Do we still need to clarify that here?

Reply:
That message has nothing to do with globbing. Any path specified as argument could be a file or a directory.

Reply:
ah. Yes they can, but this PR doesn't change that, it was true before as well. We it does change is what happens if some/dir is specified. Basically the path -> artifact rule mapping is just being relaxed here, nothing more.